### ISSUE TITLE
Union discriminator encoding

### ISSUE THREAD

**Issue by polachok** [jsgf/rust-xdr#27]
I have this union:
```
union remote_typed_param_value switch (int type) {
 case 1:
     int i;
 case 2:
     unsigned int ui;
 case 3:
     hyper l;
 case 4:
     unsigned hyper ul;
 case 5:
     double d;
 case 6:
     int b;
};
```

Generated code looks like this:
```rust
impl < Out : xdr_codec :: Write > xdr_codec :: Pack < Out > for remote_typed_param_value { fn pack ( & self , out : & mut Out ) -> xdr_codec :: Result < usize > { Ok ( match self { & remote_typed_param_value :: Const1 ( ref val ) => 1i64 . pack ( out ) ? + val . pack ( out ) ? , & remote_typed_param_value :: Const2 ( ref val ) => 2i64 . pack ( out ) ? + val . pack ( out ) ? , & remote_typed_param_value :: Const3 ( ref val ) => 3i64 . pack ( out ) ? + val . pack ( out ) ? , & remote_typed_param_value :: Const4 ( ref val ) => 4i64 . pack ( out ) ? + val . pack ( out ) ? , & remote_typed_param_value :: Const5 ( ref val ) => 5i64 . pack ( out ) ? + val . pack ( out ) ? , & remote_typed_param_value :: Const6 ( ref val ) => 6i64 . pack ( out ) ? + val . pack ( out ) ? , } ) } }
```
Discriminator (type) is encoded with i64, which seems incorrect.

RFC1014:
```

           0   1   2   3
         +---+---+---+---+---+---+---+---+
         |  discriminant |  implied arm  |          DISCRIMINATED UNION
         +---+---+---+---+---+---+---+---+
         |<---4 bytes--->|
```

**Comment by jsgf** [comment]
That does seem wrong. Is it a regression?

**Comment by polachok** [comment]
I don't know, haven't used unions before

**Comment by jsgf** [comment]
Yeah, the pack is packing i64, but the unpack is doing i32.

**Comment by polachok** [comment]
Could you release the fix to crates.io?

**Comment by jsgf** [comment]
It's still in WIP - I haven't tested it much. I was going to add at least a unit test before shipping.

**Comment by jsgf** [comment]
Should be fixed in 0.4.4 - I didn't get to adding tests, but it can't be worse than before.
